internal/provider/provider.go

```diff
@@ -337,3 +358,91 @@ func (p *btpcliProvider) DataSources(ctx context.Context) []func() datasource.Da
 		newWhoamiDataSource,
 	}, betaDataSources...)
 }
+
+func determineAuthFlow(config providerData, idToken string) string {
+	if len(idToken) > 0 {
+		return idTokenFlow
+	} else if !config.TLSClientKey.IsNull() && !config.TLSClientCertificate.IsNull() {
```

Comment:
What if only one of the two is set? Should we really assume the userPasswordFlow in that case?

Reply:
I would say so, as default case